### PR TITLE
hv: instr_emul: fix 'Parameter indexing array too big at call'

### PR DESCRIPTION
**Pull request by shiqingg**
- explicitly declare the array size of 'size2mask[]'
  otherwise, the static checking tool treats the array size to be 4, so
that it complains when accessing index '4' or '8'

- unify the comments style and make sure the line length less than 80

Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>
Reviewed-by: Junjie Mao <junjie.mao@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 1]
```diff
@@ -1312,7 +1312,8 @@ emulate_stack_op(struct vcpu *vcpu, uint64_t mmio_gpa, struct vie *vie,
 		pr_err("TODO: inject ss exception");
 	}
 
-	if (vie_alignment_check(paging->cpl, size, cr0, rflags, stack_gla) != 0) {
+	if (vie_alignment_check(paging->cpl, size, cr0, rflags, stack_gla)
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: Do you expect the  API return bool? But it is not,

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 2]
```diff
@@ -1545,7 +1546,8 @@ vmm_emulate_instruction(struct vcpu *vcpu, uint64_t gpa, struct vie *vie,
 }
 
 int
-vie_alignment_check(uint8_t cpl, uint8_t size, uint64_t cr0, uint64_t rf, uint64_t gla)
+vie_alignment_check(uint8_t cpl, uint8_t size, uint64_t cr0, uint64_t rf,
+	uint64_t gla)
 {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: What is changed?

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 3]
```diff
@@ -1589,8 +1591,8 @@ vie_size2mask(uint8_t size)
 
 int
 vie_calculate_gla(enum vm_cpu_mode cpu_mode, enum cpu_reg_name seg,
-	struct seg_desc *desc, uint64_t offset, uint8_t length, uint8_t addrsize,
-	uint32_t prot, uint64_t *gla)
+	struct seg_desc *desc, uint64_t offset, uint8_t length,
+	uint8_t addrsize, uint32_t prot, uint64_t *gla)
 {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: format change only?

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 4]
```diff
@@ -233,7 +233,7 @@ static enum cpu_reg_name gpr_map[16] = {
 	CPU_REG_R15
 };
 
-static uint64_t size2mask[] = {
+static uint64_t size2mask[9] = {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: Did u see the difference with gcc？

File: hypervisor/arch/x86/guest/instr_emul.c (hunk 3 above)
Comment (shiqingg) [reply hypervisor/arch/x86/guest/instr_emul.c]: Yes, what I changed here is to make sure that the line length is less than 80.

File: hypervisor/arch/x86/guest/instr_emul.c (hunk 2 above)
Comment (shiqingg) [reply hypervisor/arch/x86/guest/instr_emul.c]: What I changed here is to make sure that the line length is less than 80.

File: hypervisor/arch/x86/guest/instr_emul.c (hunk 1 above)
Comment (shiqingg) [reply hypervisor/arch/x86/guest/instr_emul.c]: vie_alignment_check will return either 0 or 1.
What I changed here is to make sure that the line length is less than 80.

File: hypervisor/arch/x86/guest/instr_emul.c (hunk 4 above)
Comment (shiqingg) [reply hypervisor/arch/x86/guest/instr_emul.c]: Without explicitly declaring the array size of 'size2mask[]', MISRAC will treat the array size as 4.

**Comment by lijinxia** [comment]
OK to verify

**Pull request merged**